Core/Registry/RegistryManager.cs

```diff
@@ -10,13 +11,17 @@
 
 namespace CKAN
 {
-    public class RegistryManager
+    public class RegistryManager : IDisposable
     {
         private static readonly Dictionary<string, RegistryManager> singleton =
             new Dictionary<string, RegistryManager>();
```

Comment:
First off, the formal Singleton pattern is about a class that has a public `getInstance` method which always returns a single instantiation of the pattern it wraps, so nothing about that name is helpful. If I read `static readonly` I already know that it is "initialized once and not scoped to an object reference" so you don't need to name the variable in a way describing it, but instead the purpose it serves. This is just personal opinion but I'd call this `registryCache`.